public/pages/AnomalyCharts/utils/anomalyChartUtils.ts

```diff
@@ -265,36 +271,148 @@ export const getAnomaliesHeatmapData = (
     numAnomalyGrades.push(numAnomalyGradesForEntity);
   });
 
-  const plotTimes = timeWindows.map((timeWindow) => timeWindow.endDate);
-  const plotData =
-    //@ts-ignore
-    {
-      x: plotTimes.map((timestamp) =>
-        moment(timestamp).format(HEATMAP_X_AXIS_DATE_FORMAT)
-      ),
-      y: entityValues,
-      z: maxAnomalyGrades,
-      colorscale: ANOMALY_HEATMAP_COLORSCALE,
-      //@ts-ignore
-      zmin: 0,
-      zmax: 1,
-      type: 'heatmap',
-      showscale: false,
-      xgap: 2,
-      ygap: 2,
-      opacity: 1,
-      text: numAnomalyGrades,
-      hovertemplate:
-        '<b>Time</b>: %{x}<br>' +
-        '<b>Max anomaly grade</b>: %{z}<br>' +
-        '<b>Anomaly occurrences</b>: %{text}' +
-        '<extra></extra>',
-      cellTimeInterval: timeWindows[0].endDate - timeWindows[0].startDate,
-    } as PlotData;
+  const plotTimes = timeWindows.map((timeWindow) => timeWindow.startDate);
+  const plotTimesInString = plotTimes.map((timestamp) =>
+    moment(timestamp).format(HEATMAP_X_AXIS_DATE_FORMAT)
+  );
+  const cellTimeInterval = timeWindows[0].endDate - timeWindows[0].startDate;
+  const plotData = buildHeatmapPlotData(
+    plotTimesInString,
+    entityValues,
+    maxAnomalyGrades,
+    numAnomalyGrades,
+    cellTimeInterval
+  );
   const resultPlotData = sortHeatmapPlotData(plotData, sortType, displayTopNum);
   return [resultPlotData];
 };
 
+const buildHeatmapPlotData = (
+  x: any[],
+  y: any[],
+  z: any[],
+  text: any[],
```

Comment:
I would prefer keeping z/text because that method is for building heatmap plot data, it should be coupled with `PlotData` attribute. Next time if we has another use case using heatmap chart, but not with maxAnomalyGrades/numAnomalyGrades, signature of this method can still be appropriate.